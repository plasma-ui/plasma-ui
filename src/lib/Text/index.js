import styled from '@emotion/styled';
import propTypes from '@styled-system/prop-types';
import {
  space,
  layout,
  color,
  typography,
  background,
  border,
  position,
  shadow,
  compose
} from 'styled-system';

const Text = styled('p')(
  {
    boxSizing: 'border-box'
  },
  compose(
    space,
    layout,
    color,
    typography,
    background,
    border,
    position,
    shadow
  )
);

Text.propTypes = {
  ...propTypes.space,
  ...propTypes.layout,
  ...propTypes.color,
  ...propTypes.typography,
  ...propTypes.background,
  ...propTypes.border,
  ...propTypes.position,
  ...propTypes.shadow
};

export default Text;
