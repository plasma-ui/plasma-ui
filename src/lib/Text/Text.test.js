import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Text from "./";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Text>This is a text</Text>, div);
});

it("renders correctly with matching snapshot", () => {
  const tree = renderer.create(<Text />).toJSON();
  expect(tree).toMatchSnapshot();
});
