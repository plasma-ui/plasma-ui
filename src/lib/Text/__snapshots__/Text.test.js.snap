// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with matching snapshot 1`] = `
<p
  className="css-1l4zk0c-Text"
/>
`;
